frontend/src/components/GCollapsableItems.vue

```diff
@@ -0,0 +1,158 @@
+<!--
+SPDX-FileCopyrightText: 2023 SAP SE or an SAP affiliate company and Gardener contributors
+
+SPDX-License-Identifier: Apache-2.0
+-->
+
+<template>
+  <div
+    v-if="items.length || !hideEmpty"
+    class="d-flex align-center collapsable-items-wrapper"
+  >
+    <template v-if="!expanded">
+      <slot
+        name="collapsed"
+        :item-count="itemCount"
+      >
+        {{ itemCount }}
+        {{ itemCount === 1 ? itemName : (itemPlural ? itemPlural : `${itemName}s`) }}
+      </slot>
+      <g-collapsable-items-button
+        class="collapsable-items-button"
+        :expanded="expanded"
+        @click="toggleExpanded"
+      />
+    </template>
+    <template v-else>
+      <template v-if="!items.length">
+        <slot name="noItems" />
+        <g-collapsable-items-button
+          class="collapsable-items-button"
+          :expanded="expanded"
+          @click="toggleExpanded"
+        />
+      </template>
+      <div
+        v-else
+        class="d-flex"
+        :class="noWrap ? 'flex-nowrap' : 'flex-wrap'"
+      >
+        <div
+          v-for="(item, i) in items"
+          :key="i"
+          class="d-flex align-center"
+        >
+          <slot
+            name="item"
+            :item="item"
+          />
+          <g-collapsable-items-button
+            v-if="i === items.length - 1"
+            class="collapsable-items-button"
+            :expanded="expanded"
+            @click="toggleExpanded"
+          />
+        </div>
+      </div>
+    </template>
+  </div>
+</template>
+
+<script setup>
+
+import {
+  toRefs,
+  ref,
+  computed,
+  inject,
+} from 'vue'
+
+import GCollapsableItemsButton from './GCollapsableItemsButton.vue'
+
+const props = defineProps({
+  items: {
+    type: Array,
+    default: () => [],
+  },
+  uid: {
+    type: String,
+    required: false,
+  },
+  // key for provided state
+  // If state not provided internal state is used
+  injectKey: {
+    type: String,
+    required: false,
+  },
+  hideEmpty: {
+    type: Boolean,
+    default: false,
+  },
+  itemName: {
+    type: String,
+    default: 'Item',
+  },
+  itemPlural: {
+    type: String,
+    required: false,
+  },
+  noWrap: {
+    type: Boolean,
+    default: false,
+  },
+})
+
+const { items, itemName, itemPlural, noWrap } = toRefs(props)
+const internalExpanded = ref(false)
+const expandedItems = inject(props.injectKey, undefined)
+
+const itemCount = computed(() => {
+  return props.items.length
+})
+
+const expanded = computed({
+  get () {
+    if (expandedItems) {
+      return expandedItems[props.uid] ?? expandedItems.default
+    }
+    return internalExpanded.value
+  },
+  set (value) {
+    if (expandedItems) {
+      expandedItems[props.uid] = value
```

Comment:
you could delete the key if `!value`

Reply:
if explicitly set to `false` this overrides the default state. Only if the default state is changed, I cleanup the keys. At least this is how it is implemented right now. IDK if this is the right way, though